src/edgepi/calibration/edgepi_eeprom.py

```diff
@@ -253,13 +256,29 @@ def write_memory(self, data: bytes):
         Return:
             N/A
         """
-        data_serialized = json.dumps(self.data_list) + data
-        mem_start = EdgePiMemoryInfo.USER_SPACE_START_BYTE.value + \
-            EdgePiMemoryInfo.BUFF_START.value
+        data_serialized = list(data)
+        used_mem = [(len(data_serialized)>>8)&0xFF, len(data_serialized)&0xFF]
+
+        mem_start = EdgePiMemoryInfo.USER_SPACE_START_BYTE.value +\
+                    EdgePiMemoryInfo.BUFF_START.value
+
         self.__parameter_sanity_check(mem_start, len(data_serialized), True)
+        self.log.debug(f"write_memory: length of data {data_serialized}\n{used_mem}")
+
+        # time sleep of 0.002 sec need for consecutive i2c writes/read
+        self.__byte_write_register(EdgePiMemoryInfo.USER_SPACE_START_BYTE.value, used_mem[0])
+        time.sleep(0.002)
```

Comment:
why this specific sleep delay?

Reply:
i2c throws an OSerror when the write/read is executed consecutively without delay. 0.002 second is chosen since 0.001 sec delay still threw the error